openpower/patches/vesnin-patches/hostboot/hostboot-0002-Fix-POS-and-SPD-VPD_REC_NUM.patch

```diff
@@ -0,0 +1,28 @@
+From 1dcf087af57a93ec7f24aa8be73e7ff0ffad22ca Mon Sep 17 00:00:00 2001
+From: Artem Senichev <a.senichev@yadro.com>
+Date: Tue, 13 Mar 2018 15:03:41 +0300
+Subject: [PATCH] Fix POS and SPD-VPD_REC_NUM
+
+Originally created by MSI (S188)
+
+Signed-off-by: Artem Senichev <a.senichev@yadro.com>
+---
+ src/usr/targeting/common/Targets.pm | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/src/usr/targeting/common/Targets.pm b/src/usr/targeting/common/Targets.pm
+index 35f08428c..4eaba16e7 100644
+--- a/src/usr/targeting/common/Targets.pm
++++ b/src/usr/targeting/common/Targets.pm
+@@ -744,7 +744,7 @@ sub processMcs
+                         $self->setAttribute($dimm,"MBA_PORT",$port_num);
+                         $self->setAttribute($dimm,"MBA_DIMM",$dimm_num);
+ 
+-                        my $aff_pos=16*$proc+$mcs*$self->{MAX_MCS}+4*$mba+
++                        my $aff_pos=64*$proc+$mcs*$self->{MAX_MCS}+4*$mba+
```

Comment:
Was there a reason you needed numbers this high?  Physically there can only be 16 dimms behind each processor.

Reply:
When bringup this 4Socket 128DIMMs system, we hit the following VPD/SPD issue and suspected it might be related to the VPD_REC_NUM numbering algorithm issue then caused the SPD cache overlap for the 4 sockets case. (attached the default output VPD_REC_NUM_ORG.txt, you would see some duplicated numbering starting from 3rd CPU's DIMMs like VPD_REC_NUM=32..47)
[VPD_REC_NUM_ORG.txt](https://github.com/open-power/op-build/files/1858672/VPD_REC_NUM_ORG.txt)
```

 16.23577|================================================
 16.23577|Error reported by vpd (0x1D00)
 16.24571|  Invalid Basic Memory Type
 16.24572|  ModuleId   0x61 VPD::VPD_SPD_GET_KEYWORD_VALUE
 16.24572|  ReasonCode 0x1d07 VPD::VPD_INVALID_BASIC_MEMORY_TYPE
 16.24572|  UserData1  Basic Memory Type (Byte 2) : 0x00000000000000ff
 16.24573|  UserData2  Keyword Requested : 0x0000000000000025
```
 
So we simply tried to jump the $aff_pos offset from 16 to 64 for each Proc and the issue was fixed/gone. 
[VPD_REC_NUM_MOD.txt](https://github.com/open-power/op-build/files/1858673/VPD_REC_NUM_MOD.txt)

Reply:
BTW, we've also checked a booted PNOR dump with 128DIMMs installed, seems like all the 128 DIMM SPD cache regions can be updated properly within the DJVPD partition range. (0x91000-0xD9000)
[DJVPD_Dump.bin.zip](https://github.com/open-power/op-build/files/1864297/DJVPD_Dump.bin.zip)